### PR TITLE
Fine tune eventing CRDs and Makefile

### PR DESCRIPTION
**Pull request by sayanh**
**Description**

Changes proposed in this pull request:

- Fine tune CRDs and makefile
- Use correct kubebuilder tags to generate consistent CRDs

**Related issue(s)**

See https://github.com/kyma-project/kyma/issues/11418

### CODE REVIEW

File: components/eventing-controller/api/v1alpha1/subscription_types.go [hunk 1]
```diff
@@ -4,14 +4,20 @@ import (
...
-// Webhook defines the Webhook called by an active subscription in BEB
+// WebhookAuth defines the Webhook called by an active subscription in BEB
 type WebhookAuth struct {
-	Type         string   `json:"type,omitempty"`
-	GrantType    string   `json:"grantType"`
-	ClientId     string   `json:"clientId"`
-	ClientSecret string   `json:"clientSecret"`
-	TokenUrl     string   `json:"tokenUrl"`
-	Scope        []string `json:"scope,omitempty"`
+	// +optional
```
Comment (k15r) [inline components/eventing-controller/api/v1alpha1/subscription_types.go:9]: could we also add descriptions to those properties?

Comment (sayanh) [reply components/eventing-controller/api/v1alpha1/subscription_types.go:10]: The comment [here](https://github.com/kyma-project/kyma/pull/11447/files/12bb85ec4de2c7228b6022340755a1f85ecaa6ae#diff-15ef6a41dff50cd9aa8cd717dba2729f245f21b5ceeae3527b268ec7804256d2R7) is explanatory hence left it intentionally to make it less verbose. I added them anyway.

**Review by pxsalehi — APPROVED** [review]

**Review by alexandra-simeonova — APPROVED** [review]

**Review by a-thaler — APPROVED** [review]

**Review by werdes72 — APPROVED** [review]

**Review by alexandra-simeonova — DISMISSED** [review]

**Review by a-thaler — DISMISSED** [review]

**Review by werdes72 — DISMISSED** [review]

**Review by pxsalehi — APPROVED** [review]

**Review by alexandra-simeonova — APPROVED** [review]

**Review by a-thaler — APPROVED** [review]

**Review by werdes72 — APPROVED** [review]

**Review by alexandra-simeonova — APPROVED** [review]

**Review by werdes72 — APPROVED** [review]

**Review by pxsalehi — APPROVED** [review]

**Review by a-thaler — APPROVED** [review]

**Pull request merged**